numpy/linalg/linalg.py

```diff
@@ -563,26 +562,17 @@ def matrix_power(M, n):
     LinAlgError
         If the matrix is not numerically invertible.
```

Comment:
Should mention that being non-square does this too.

Also, this comment is sort of false - this only happens if it's non-invertable and `n` is negative.